bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/BookieShell.java

```diff
@@ -2050,6 +2059,55 @@ protected void readLedgerIndexEntries(long ledgerId) throws IOException {
     }
 
     /**
+     * Read ledger index entries
+     * 
+     * @param ledgerId
+     * @return
+     * @throws IOException if failed to init KeyValueStorageRocksDB
+     */
+	protected int readLedgerIndexEntriesFromDbLedgerStorage(long ledgerId) throws IOException {
+
+		ServerConfiguration conf = new ServerConfiguration(bkConf);
+		LedgerDirsManager ledgerDirsManager = new LedgerDirsManager(bkConf, bkConf.getLedgerDirs());
+		String ledgersPath = ledgerDirsManager.getAllLedgerDirs().get(0).toString();
+
+		EntryLocationIndex entryLocationIndex = null;
+
+		try {
+			entryLocationIndex = new EntryLocationIndex(conf,
+					(path, dbConfigType, conf1) -> new KeyValueStorageRocksDB(path, DbConfigType.Small, conf1, true),
+					ledgersPath, NullStatsLogger.INSTANCE);
+		} catch (IOException e) {
+			System.err.printf("ERROR: initializing rocksDb storage %s", e.getMessage());
+			return -1;
+		}
+		try {
+			long lastEntryId = entryLocationIndex.getLastEntryInLedger(ledgerId);
+			for (long curEntry = 0; curEntry <= lastEntryId; curEntry++) {
+				try {
+					long offset = entryLocationIndex.getLocation(ledgerId, curEntry);
+					if (offset <= 0) {
+						// no-more entry found
+						System.out.println("entry " + curEntry + "\t:\tN/A");
+						break;
+					}
+					long entryLogId = offset >> 32L;
+					long pos = offset & 0xffffffffL;
+					System.out.println("entry " + curEntry + "\t:\t(log: " + entryLogId + ", pos: " + pos + ")");
+				} catch (Exception e) {
+					System.err.println("ERROR: while getting location for entry " + ledgerId + ", " + curEntry + ", "
```

Comment:
A single bookie is not required to have all the entries for a particular ledger. 

If you have `e=3 w=2 a=2` for example, each bookie will be missing 1 entry out of 3, we shouldn't break the loop at that point.

Also, the first entry might be very after 0. Eg: when a bookie crashes (or times out), the client will keep using the same ledger and form a new ensemble by removing the old bookie and picking a new one. 
The new bookie will only have entries from that point on.